pom.xml

```diff
@@ -18,7 +18,7 @@
 
     <properties>
         <java.version>11</java.version>
-        <java.test.version>17</java.test.version>
+        <java.test.version>11</java.test.version>
```

Comment:
What's the reason for going back here? I suppose it will cause some existing tests to fail.

Reply:
IntelliJ didn't respect the different JVM for test as for the main code. So it was seeing parser/compiler errors on the multi-line string literals and refusing to run the tests.

Reply:
I see. So indeed in IDEs, the configuration needs to be set to 17 (as most (all?) of them don't differentiate between different levels for main and test code). This was a once-off config for me in Eclipse (and probably I'll have to redo it when updating the project config). That felt acceptable to me, given the advantages of e.g. being able to specify YAML in test code inline, without any escaping or line concatenation. If we want to change it to 11, some existing tests will need to be adjusted. My personal vote would be on keeping 11 for main and 17 for testing.

Reply:
Let me see if I can make [IDEA-85478](https://youtrack.jetbrains.com/issue/IDEA-85478) work

Reply:
With some fiddling that seems to work. Had to switch to using the maven wrapper, rather than IDEA's built-in maven (which was too old a version) and also tick "Create separate modules for production and test roots" in `Settings > Maven > Importing`, and finally configure the right JVM for the test.